Use swc over ts-node causes errors with Alpine Linux images

## Current Behavior
After upgrading to `13.6.1` our `lint`  (Gitlab) pipeline fails due to:
```
Failed to load plugin '@nrwl/nx' declared in 'packages/utils/.eslintrc.json » ../../.eslintrc.json#overrides[0]': Cannot find module '@swc/core-linux-x64-musl'
```

Command used:
`npx nx affected:lint --base=origin/$CI_MERGE_REQUEST_TARGET_BRANCH_NAME --head=HEAD`

Seems to be related to:
- https://github.com/nrwl/nx/pull/8694

(13.5.3 and 13.6.0 work fine)

*N.B.* 
Actively installing the missing package in the pipeline seems to fix it:
```yaml
- npm ci --cache .npm --prefer-offline --legacy-peer-deps --no-progress --no-optional --no-audit
- npm i -D @swc/core-linux-x64-musl --legacy-peer-deps
```    
*N.B.2*
Can't install it locally (MacOS), since: 
```
Unsupported platform for @swc/core-linux-x64-musl@1.2.135: wanted {"os":"linux","arch":"x64"} (current: {"os":"darwin","arch":"x64"})
```

*N.B.3*
- We are using an ALPINE Linux image.
- https://swc.rs/docs/getting-started#supported-binaries
  
![Screenshot 2022-01-28 at 15 42 37](https://user-images.githubusercontent.com/4863062/151566719-8faddce7-cea1-4d3f-be77-c3d92a509861.png)

### Environment

`NODE_IMAGE: node:16.7.0-alpine3.14`
(same on 16.13.2)

```
Running with gitlab-runner 14.7.0 (98daeee0)
  on ...
  feature flags: FF_USE_FASTZIP:true
Preparing the "kubernetes" executor
00:00
Using Kubernetes namespace: ...-gitlabhost-runner
Using Kubernetes executor with image ....dkr.ecr.eu-west-1.amazonaws.com/node:16.7.0-alpine3.14 ...
Using attach strategy to execute scripts...
```

```
$ npx nx report
 >  NX   Report complete - copy this into the issue template
   Node : 16.7.0
   OS   : linux x64
   npm  : 7.20.3
   
   nx : 13.6.1
   @nrwl/angular : 13.6.1
   @nrwl/cli : 13.6.1
   @nrwl/cypress : 13.6.1
   @nrwl/detox : undefined
   @nrwl/devkit : 13.6.1
   @nrwl/eslint-plugin-nx : 13.6.1
   @nrwl/express : undefined
   @nrwl/jest : 13.6.1
   @nrwl/js : undefined
   @nrwl/linter : 13.6.1
   @nrwl/nest : undefined
   @nrwl/next : undefined
   @nrwl/node : undefined
   @nrwl/nx-cloud : undefined
   @nrwl/react : undefined
   @nrwl/react-native : undefined
   @nrwl/schematics : undefined
   @nrwl/storybook : 13.6.1
   @nrwl/tao : 13.6.1
   @nrwl/web : undefined
   @nrwl/workspace : 13.6.1
   typescript : 4.5.5
   rxjs : 7.5.2
   ---------------------------------------
   Community plugins:
   	 @angular/animations: 13.2.0
   	 @angular/cdk: 13.2.0
   	 @angular/common: 13.2.0
   	 @angular/compiler: 13.2.0
   	 @angular/core: 13.2.0
   	 @angular/forms: 13.2.0
   	 @angular/material: 13.2.0
   	 @angular/platform-browser: 13.2.0
   	 @angular/platform-browser-dynamic: 13.2.0
   	 @angular/router: 13.2.0
   	 @angular-devkit/build-angular: 13.2.0
   	 @angular/cli: 13.2.0
   	 @angular/compiler-cli: 13.2.0
   	 @angular/language-service: 13.2.0
   	 @compodoc/compodoc: 1.1.18
   	 @jscutlery/semver: 2.17.6
   	 @nx-plus/docusaurus: 12.2.0
   	 @storybook/angular: 6.4.15
   	 @testing-library/angular: 11.0.2
   	 @twittwer/compodoc: 1.6.6
   	 ng-samurai: 3.0.0
```

@see: https://nrwlcommunity.slack.com/archives/CMFKWPU6Q/p1643368191765079

Same issue on azure DevOps and no issue locally on macOS:

```
Failed to load plugin '@nrwl/nx' declared in 'apps/smc/mf-list/.eslintrc.json » ../../../.eslintrc.js#overrides[1]': Cannot find module '@swc/core-linux-x64-gnu'
```

Nx Report:
![image](https://user-images.githubusercontent.com/954509/152102565-13088a5b-2d3e-42ca-8e78-c38fc998ea1c.png)

--- a/package.json
+++ b/package.json
@@ -257,6 +257,9 @@
     "yargs-parser": "20.0.0",
     "zone.js": "~0.11.4"
   },
+  "optionalDependencies": {
+    "@swc/core-linux-musl": "~1.2.42"
+  },
   "author": "Victor Savkin",
   "license": "MIT",
   "config": {

--- a/packages/eslint-plugin-nx/package.json
+++ b/packages/eslint-plugin-nx/package.json
@@ -39,5 +39,8 @@
     "chalk": "4.1.0",
     "confusing-browser-globals": "^1.0.9",
     "tsconfig-paths": "^3.9.0"
+  },
+  "optionalDependencies": {
+    "@swc/core-linux-musl": "^1.2.42"
   }
 }

